### PR TITLE
cleanup work for appium 1.5

### PR DESCRIPTION
**Pull request by Jonahss**
@imurchie

### CODE REVIEW

File: lib/simulator-xcode-6.js [hunk 1]
```diff
@@ -184,18 +187,33 @@ class SimulatorXcode6 {
...
+    let checkBooted = async function () {
+      let stat = await this.stat();
+      return new B((resolve, reject) => {
+        if (stat.state === 'Booted') {
+          resolve();
+        } else {
+          reject();
+        }
+      });
+    }.bind(this);
```
Comment (jlipps) [inline lib/simulator-xcode-6.js]: why not just `let checkBooted = async () => {` so you don't have to bind?

File: lib/simulator-xcode-6.js [hunk 2]
```diff
@@ -184,18 +187,33 @@ class SimulatorXcode6 {
...
+
+    await retryInterval(parseInt(BOOTING_TIMOUT/BOOTED_POLLING_INTERVAL), BOOTED_POLLING_INTERVAL, () => {
```
Comment (jlipps) [inline lib/simulator-xcode-6.js]: i think by our conventions this should be an async function and `return await checkBooted();`

File: lib/simulator-xcode-6.js [hunk 3]
```diff
@@ -184,18 +187,33 @@ class SimulatorXcode6 {
...
+    // start simulator
+    let simulatorApp = path.resolve(await getXcodePath(), 'Applications', 'Simulator.app');
+    let args = [simulatorApp, '--args', '-CurrentDeviceUDID', this.udid];
+    log.info(`starting simulator with command: open ${args.join(' ')}`);
+    await exec('open', args, {timeout: STARTUP_TIMEOUT});
+
+    //wait for it to be 'booted'
+    let checkBooted = async function () {
+      let stat = await this.stat();
+      return new B((resolve, reject) => {
```
Comment (jlipps) [inline lib/simulator-xcode-6.js]: what is the point of creating a promise here? this whole function can go away and below you can just have as the anonymous async function:

```js
async () => {
  let stat = await this.stat();
  if (stat.state !== 'Booted') {
    throw new Error("State wasn't booted");
  }
}
```

File: lib/simulator-xcode-6.js [hunk 4]
```diff
@@ -208,13 +226,18 @@ class SimulatorXcode6 {
     log.debug('Attempting to launch and quit the simulator, to create directory structure');
     log.debug(`Will launch with Safari? ${safari}`);
 
-    await instrumentsUtil.quickLaunch(this.udid, safari ? 'com.apple.mobilesafari' : undefined);
+    if (safari) {
+      await instrumentsUtil.quickLaunch(this.udid, safari ? 'com.apple.mobilesafari' : undefined);
+    } else {
+      this.run();
```
Comment (jlipps) [inline lib/simulator-xcode-6.js]: isn't this an async function? shouldn't it be awaited?

**Comment by imurchie** [comment]
I've addressed these issues in an upcoming PR. :)

**Pull request merged**